AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m

```diff
@@ -96,7 +96,8 @@ + (void)performDbTokenAsyncOperationWithHttpClient:(id<MSHttpClientProtocol>)htt
           }
 
           // Create token result object.
-          MSTokenResult *tokenResult = [[MSTokenResult alloc] initWithDictionary:jsonDictionary[kMSTokens][0]];
+          // FIXME: we should eventually validate further the payload (we might not get the dictionary we expect).
+          MSTokenResult *tokenResult = [[MSTokenResult alloc] initWithDictionary:(NSDictionary *)jsonDictionary[kMSTokens][0]];
```

Comment:
If there is only one token you might want to rename `kMSTokens` to `kMSToken` :)

Reply:
No, we retrieve tokens one by one, but the API can return multiple tokens. The constant here is the name of the property in the payload, it's `tokens`.

Reply:
Oh so it's an array of tokens and we get the first one. I would quickly check the count on the array before accessing the first one. Reading the FIXME looks like you have a task already to fix this though.